### PR TITLE
Update Deneb to 1.4.0-beta.2 (devnet-9)

### LINKED ISSUE #4748: Implement fork choice proposer boost uniqueness
## Description

The latest spec release includes the change to apply proposer boost only to the first block: https://github.com/ethereum/consensus-specs/pull/3352.

I think we should make this change on the Deneb branch, and roll it out gradually when we merge Deneb to `stable`. This will allow us to pass the latest spec tests on the Deneb branch.

## Steps to resolve

We can also delete the temporary mitigation we added for the RPC unbundling attack, either as part of the same PR, or in a follow-up.

### PR DESCRIPTION
**Pull request by dapplion**
## Proposed Changes

- Implements https://github.com/ethereum/consensus-specs/pull/3499
- Implements https://github.com/ethereum/consensus-specs/pull/3352
- Updates test vectors to `1.4.0-beta.2-hotfix`
- Updates fork choice test runner to be blob-aware. See spec here: https://github.com/ethereum/consensus-specs/tree/master/tests/formats/fork_choice#on_block-execution-step.

## Additional Info

- Config values for non-mainnet networks are set to mainnet as a TBA
- Closes #4748

### CODE REVIEW

**Comment by michaelsproul** [comment]
@dapplion I'm going to get this PR passing the tests and merge it :+1:

**Comment by michaelsproul** [comment]
This is still WIP, I need to do some more massaging of the EF fork choice tests.

**Comment by michaelsproul** [comment]
I've got the EF tests passing now. Ready for review

File: consensus/types/src/chain_spec.rs [hunk 1]
```diff
@@ -686,6 +688,8 @@ impl ChainSpec {
             config_name: None,
             max_committees_per_slot: 4,
             target_committee_size: 4,
+            min_per_epoch_churn_limit: 2,
```
Comment (ethDreamer) [inline consensus/types/src/chain_spec.rs:691]: nice catch

Comment (ethDreamer) [inline consensus/types/src/chain_spec.rs:691]: nice catch

**Review by ethDreamer — APPROVED** [review]

File: consensus/types/src/chain_spec.rs (hunk 1 above)
Comment (michaelsproul) [reply consensus/types/src/chain_spec.rs:691]: I was scratching my head for a while over this one, wondering why so many tests were failing on the new spec :sweat_smile:

File: testing/ef_tests/src/cases/fork_choice.rs [hunk 2]
```diff
@@ -380,16 +404,72 @@ impl<E: EthSpec> Tester<E> {
...
+                let result = self.block_on_dangerous(
+                    self.harness
+                        .chain
+                        .check_gossip_blob_availability_and_import(
+                            GossipVerifiedBlob::__assumed_valid(signed_sidecar),
+                        ),
+                )?;
+                if valid {
+                    assert!(result.is_ok());
+                }
```
Comment (jimmygchen) [inline testing/ef_tests/src/cases/fork_choice.rs:458]: do we want to check for the invalid case here, i.e. `assert_eq!(valid, result.is_ok())`?

Comment (michaelsproul) [reply testing/ef_tests/src/cases/fork_choice.rs:458]: no, because some blobs from the block may be valid even if the whole block with blobs is invalid

Comment (jimmygchen) [reply testing/ef_tests/src/cases/fork_choice.rs:458]: ah got it, thanks!

Comment (jimmygchen) [reply testing/ef_tests/src/cases/fork_choice.rs:458]: letss merge 🚀

**Pull request merged**